docs/topics/addons.rst

```diff
@@ -0,0 +1,182 @@
+.. _topics-addons:
+
+=======
+Add-ons
+=======
+
+Scrapy's add-on system is a framework which unifies managing and configuring
+components that extend Scrapy's core functionality, such as middlewares,
+extensions, or pipelines. It provides users with a plug-and-play experience in
+Scrapy extension management, and grants extensive configuration control to
+developers.
+
+
+Activating and configuring add-ons
+==================================
+
+Add-ons and their configuration live in Scrapy's
+:class:`~scrapy.addons.AddonManager`. During a :class:`~scrapy.crawler.Crawler`
+initialization the add-on manager will read a list of enabled add-ons from your
+``ADDONS`` setting.
+
+The ``ADDONS`` setting is a dict in which every key is an addon class or its
+import path and the value is its priority.
+
+This is an example where two add-ons are enabled in a project's
+``settings.py``::
+
+    ADDONS = {
+        'path.to.someaddon': 0,
+        path.to.someaddon2: 1,
+    }
+
+
+Writing your own add-ons
+========================
+
+Add-ons are (any) Python objects that include the following method:
```

Comment:
Even if this may be technically true, shouldn’t we encourage them being classes from the docs?